src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -1,214 +1,144 @@
+                /* component = */ subscriptionPanel,
+                /* tip = */ null,
+                SUBSCRIPTION_TAB_INDEX)
+          }
+          subscriptionPanel.update(data.isCurrentUserPro)
+        } else if (isSubscriptionTabPresent) {
+          tabbedPane.remove(SUBSCRIPTION_TAB_INDEX)
         }
       }
     }
   }
 
-  @RequiresEdt
-  private fun refreshSubscriptionTab(data: SubscriptionTabPanelData) {
-    val isSubscriptionTabPresent = tabbedPane.tabCount >= SUBSCRIPTION_TAB_INDEX + 1
-
-    if (data.isDotcomAccount && data.codyProFeatureFlag) {
-      if (!isSubscriptionTabPresent) {
-        tabbedPane.insertTab(
-            /* title = */ "Subscription",
-            /* icon = */ null,
-            /* component = */ subscriptionPanel,
-            /* tip = */ null,
-            SUBSCRIPTION_TAB_INDEX)
+  private fun runNewCommand(project: Project, commandId: CommandId) {
+    startNewPanel(project) { agent: CodyAgent ->
+      when (commandId) {
+        CommandId.Explain -> agent.server.commandsExplain()
+        CommandId.Smell -> agent.server.commandsSmell()
+        CommandId.Test -> agent.server.commandsTest()
       }
-      subscriptionPanel.update(data.isCurrentUserPro)
-    } else if (isSubscriptionTabPresent) {
-      tabbedPane.remove(SUBSCRIPTION_TAB_INDEX)
     }
   }
 
-  fun refreshChatToEmpty() {
-    CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
-      ApplicationManager.getApplication().invokeLater {
-        messagesPanel.removeAll()
-        chatMessageHistory.clearHistory()
-        inProgressChat.abort()
-        addWelcomeMessage()
-        activateChatTab()
-      }
-      chatId = agent.server.chatNew().get()
-    }
+  private fun startNewChat(project: Project) {
+    startNewPanel(project) { it.server.chatNew() }
   }
 
-  private fun selectChat(item: ChatState) {
-    ApplicationManager.getApplication().invokeLater {
-      messagesPanel.removeAll()
-      addWelcomeMessage()
-      val chat = HistoryService.getInstance().getChatByPanelId(item.panelId!!)
-      for (message in chat.messages) {
-        displayUsedContext(
-            message.contextFiles.map {
-              ContextMessage(Speaker.ASSISTANT, "", ContextFile(URI.create(it)))
-            })
-        addChatMessageAsComponent(ChatMessage(message.speaker!!, message.text, message.text))
-      }
-      CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
-        val model = agent.server.chatModels(ChatModelsParams(chatId!!)).get().models.first().model
-        val chatMessages = chat.messages.map { ChatMessage(it.speaker!!, it.text, it.text) }
-        if (item.replyChatId != null) {
-          val restoredId =
-              agent.server
-                  .chatRestore(ChatRestoreParams(model, chatMessages, item.replyChatId!!))
-                  .get()
-          chat.panelId = restoredId
-          chatId = restoredId
+  private fun startNewPanel(
+      project: Project,
```

Comment:
`project` is a private val of this class already - there is no need to pass it here